packages/service-library/src/servicelib/aiohttp/application_setup.py

```diff
@@ -150,26 +178,17 @@ def setup(app: web.Application):
     # TODO: resilience to failure. if this setup fails, then considering dependencies, is it fatal or app can start?
     # TODO: enforce signature as def setup(app: web.Application, **kwargs) -> web.Application
 
-    module_name = module_name.replace(".__init__", "")
-    depends = depends or []
-
-    if config_section and config_enabled:
-        raise ValueError("Can only set config_section or config_enabled but not both")
-
-    section = config_section or module_name.split(".")[-1]
-    if config_enabled is None:
-        config_enabled = f"{section}.enabled"
-    else:
-        # if passes config_enabled, invalidates info on section
-        section = None
+    module_name, depends, section, config_enabled = _parse_and_validate_arguments(
+        module_name, depends, config_section, config_enabled
+    )
 
     def _decorate(setup_func: _SetupFunc):
 
         if "setup" not in setup_func.__name__:
             logger.warning("Rename '%s' to contain 'setup'", setup_func.__name__)
 
         # metadata info
-        def setup_metadata() -> Dict[str, Any]:
+        def setup_metadata() -> SetupMetadataDict:
             return {
                 "module_name": module_name,
                 "dependencies": depends,
```

Comment:
you probably need to add some asserts in front?

Reply:
What do you mean?